typecats/tc.py

```diff
@@ -117,9 +117,9 @@ def get_default_converter():
     return _TYPECATS_DEFAULT_CONVERTER
 
 
-def register_struc_hook(*args, **kwargs):
+def register_struc_hook(typ, func, **_):
     """Use this to register cattrs structuring hooks on the internal cattrs Converter"""
-    _TYPECATS_DEFAULT_CONVERTER.register_structure_hook(*args, **kwargs)
+    _TYPECATS_DEFAULT_CONVERTER.register_structure_hook_func(lambda t: t == typ, func)
```

Comment:
I'd like to investigate this further- newer versions of cattrs support typeddicts (23.1.0 and up)  - I don't know what implications changing register_struc_hook has but I'd be wary of it